streams/src/main/java/org/apache/kafka/streams/processor/ConsumerRecordTimestampExtractor.java

```diff
@@ -0,0 +1,37 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.processor;
+
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+
+/**
+ * Retrieves built-in timestamps from Kafka messages (introduced in KIP-32: Add timestamps to Kafka message).
+ *
+ * Here, "built-in" refers to the fact that compatible Kafka producer clients automatically and
+ * transparently embed such timestamps into messages they sent to Kafka, which can then be retrieved
+ * via this timestamp extractor.
+ *
+ * Depending on whether <i>CreateTime</i> or <i>LogAppendTime</i> are used to define the built-in timestamps,
+ * using this extractor effectively provide event-time or processing-time semantics
```

Comment:
Question: Does `LogAppendTime` really provide processing-time semantics?

My understanding is:

* `CreateTime` = Kafka producer time = **event-time semantics**
* `LogAppendTime` = broker time
*  System.currentTimeMillis() = WallclockTimestampExtractor = Kafka consumer time = **processing-time semantics**